### DISCUSSION TITLE
Conversion with EPSG_RGF93_V2B_LAMBERT_93 dosen't work for Corsica

### DISCUSSION THREAD

**Discussion by Terahdon** [dvdoug/PHPCoord#43] [Q&A]
Hi,

First, thanks a lot for this library it's working great.

But I have a problem with converting coordinates in LAMBERT_93 to latitude and longitude for CORSICA.

Here is my code :
```

/**
     * @throws UnknownCoordinateReferenceSystemException
     * @throws UnknownConversionException
     */
    private function fromLAMBERT_93toLatLong(string $coordxet, string $coordyet): ?Coordinate
    {
        $crs = Projected::fromSRID(Projected::EPSG_RGF93_V2B_LAMBERT_93);
        /** @var GeographicPoint $to */
        $to = $this->convert($crs, $coordxet, $coordyet);
        $to instanceof GeographicPoint ? $coordinate = $this->coordinate((string)$to->getLatitude(), (string)$to->getLongitude(), self::LAMBERT_93) : $coordinate = null;
        return $coordinate;
    }
```
```

   private function convert(Projected $crs, string $coordxet, string $coordyet): ?Point
    {
        try {
            $from = ProjectedPoint::create(
                new Metre((float)$coordxet),
                new Metre((float)$coordyet),
                null,
                null,
                $crs
            );

            $toCRS = Geographic2D::fromSRID(Geographic2D::EPSG_RGF93_V2B);
            return $from->convert($toCRS);
        } catch (UnknownCoordinateReferenceSystemException|UnknownConversionException $e) {
            $this->logger->logError($e->getMessage());
            throw $e;
        }
    }
```

**Basically it can be reduced to :** 
```
$crs = Projected::fromSRID(Projected::EPSG_RGF93_V2B_LAMBERT_93);
        /** @var GeographicPoint $to */
        $to = $this->convert($crs, $coordxet, $coordyet);
```
```
$from = ProjectedPoint::create(
    new Metre((float)$coordxet),
    new Metre((float)$coordyet),
    null,
    null,
    $crs
);

$toCRS = Geographic2D::fromSRID(Geographic2D::EPSG_RGF93_V2B);
return $from->convert($toCRS);
```
The code is working for every coordinates I have (more than 20000) but not for the one from CORSICA
So it's weird because I use EPSG_RGF93_V2B and EPSG_RGF93_V2B_LAMBERT_93
Both have      * Extent: France - onshore and offshore, mainland and Corsica.

The coordinates comes from an official french government datafile 

coordx => '1223977.1'  coordy => '6075793.1'

the source : https://www.data.gouv.fr/fr/datasets/r/98f3161f-79ff-4f16-8f6a-6d571a80fea2

the specific line with LAMBERT_93 specified

`geolocalisation;2A0001335;1223977.1;6075793.1;2,ATLASANTE,100,IGN,BD_ADRESSE,V2.2,LAMBERT_93;2022-03-10`

I also tried all the other lambert_93 types of coordinates.

Do you have an idea from where it could come from ?
Thanks a lot if you find a little bit of time to help me.

**Comment by Terahdon** [comment]
Well it seem to work with $ignoreBoundaryRestrictions set to true. 
I'm' getting those coordinates
```
 {
  latitude: "41.944239739838"
  longitude: "8.7532823507679"
  coordinateType: "LAMBERT_93"
}

```
But I don't get why.

**Reply by dvdoug** [reply]
> Well it seem to work with $ignoreBoundaryRestrictions set to true. I'm' getting those coordinates

Ah, thanks - that will help a lot in tracking this down!

Could I just check what version of the library you're using, and also whether you're using the Europe data pack or not so I can debug with the same configuration?

**Comment by dvdoug** [comment]
Hi @Terahdon 

I've done a bit of digging, a few different things going on here.

1) I converted the point you specified using the official IGN tool Circe, and got the numbers below - which are different from the ones you have given as an example. So I would still like to confirm which version of PHPCoord(+any datapack) you're using please as the testing I've just done (master branch) does agree with Circe.
![image](https://user-images.githubusercontent.com/1571110/172003967-3d142bcb-2646-4c61-8272-d41f4216f9f7.png)

2) The reason you're having difficulty with Corsica, is because although EPSG have RGF/Lambert-93 defined to cover "France", they have the Lambert-93 conversion mechanism specified as being valid only for "France - mainland onshore". This is why PHPCoord says the conversion is out of bounds. You can see that at https://epsg.org/crs_9794/RGF93-v2b-Lambert-93.html and https://epsg.org/conversion_18085/Lambert-93.html.

I will email EPSG to point out the discrepancy and ask them if it can be fixed.

**Reply by Terahdon** [reply]
@dvdoug Yes it's my mistake, to get the result in my first comment I tested it with coordx = '1177682.07' and coordy = '6111141.87'. I mixed things up sorry.

For the numbers I gave you (coordx => '1223977.1' coordy => '6075793.1') I do get 
 {
  latitude: "41.596016636048"
  longitude: "9.27485292278"
  coordinateType: "LAMBERT_93"
}

**Reply by dvdoug** [reply]
OK, great. In that case since there's no bug in the algorithm itself, only the algorithm selection (because of the incorrect boundary) please do continue using `$ignoreBoundaryRestrictions` for the conversions you are doing.

Because the discrepancy is now explained and understood, this is safe to do so.

**Reply by dvdoug** [reply]
OK, great. In that case since PHPCoord is doing the Lambert-93 conversion correctly, and the boundary being wrong is known and understood I recommend you keep using `$ignoreBoundaryRestrictions=true`. For the conversions you are doing, it is safe to do so.

**Comment by Terahdon** [comment]
Hello @dvdoug and ty for your quick answer !


**In my composer.json =>**  `"php-coord/php-coord": "^4.6"`
    


**And probably not necessary but in my composer.lock =>**
    
```
            {
            "name": "php-coord/php-coord",
            "version": "v4.6.1",
            "source": {
                "type": "git",
                "url": "https://github.com/dvdoug/PHPCoord.git",
                "reference": "f3ff8a09cfb12ba6d232b8147e0fb1f7ac593720"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/dvdoug/PHPCoord/zipball/f3ff8a09cfb12ba6d232b8147e0fb1f7ac593720",
                "reference": "f3ff8a09cfb12ba6d232b8147e0fb1f7ac593720",
                "shasum": ""
            },
            "require": {
                "php": "^7.4||^8.0",
                "symfony/polyfill-php80": "^1.18"
            },
            "require-dev": {
                "ext-json": "*",
                "ext-sqlite3": "*",
                "friendsofphp/php-cs-fixer": "^3.0",
                "nikic/php-parser": "^4.10",
                "php-coord/datapack-africa": "^1.0",
                "php-coord/datapack-antarctic": "^1.0",
                "php-coord/datapack-arctic": "^1.0",
                "php-coord/datapack-asia": "^1.0",
                "php-coord/datapack-europe": "^1.0",
                "php-coord/datapack-northamerica": "^1.0",
                "php-coord/datapack-oceania": "^1.0",
                "php-coord/datapack-southamerica": "^1.0",
                "phpunit/phpunit": "^9.5.10"
            },
            "suggest": {
                "php-coord/datapack-africa": "High-accuracy addon datapack for Africa",
                "php-coord/datapack-antarctic": "High-accuracy addon datapack for the Antartic",
                "php-coord/datapack-arctic": "High-accuracy addon datapack for the Arctic",
                "php-coord/datapack-asia": "High-accuracy addon datapack for Asia",
                "php-coord/datapack-europe": "High-accuracy addon datapack for Europe",
                "php-coord/datapack-northamerica": "High-accuracy addon datapack for North America",
                "php-coord/datapack-oceania": "High-accuracy addon datapack for Oceania",
                "php-coord/datapack-southamerica": "High-accuracy addon datapack for South America"
            },
            "type": "library",
            "extra": {
                "branch-alias": {
                    "dev-master": "4.x-dev"
                }
            },
            "autoload": {
                "psr-4": {
                    "PHPCoord\\": "src/"
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "(MIT and proprietary)"
            ],
            "authors": [
                {
                    "name": "Doug Wright",
                    "role": "Developer"
                }
            ],
            "description": "PHPCoord is a PHP library to aid in handling coordinates. It can convert coordinates for a point from one system to another and also calculate distance between points.",
            "homepage": "https://www.phpcoord.net/",
            "keywords": [
                "UTM",
                "coord",
                "ed50",
                "epsg",
                "gda",
                "geo",
                "grid ref",
                "latitude",
                "longitude",
                "map",
                "nad",
                "nzmg",
                "osgb36",
                "projection",
                "wgs84"
            ],
            "support": {
                "issues": "https://github.com/dvdoug/PHPCoord/issues",
                "source": "https://github.com/dvdoug/PHPCoord/tree/v4.6.1"
            },
            "funding": [
                {
                    "url": "https://github.com/dvdoug",
                    "type": "github"
                }
            ],
            "time": "2022-01-29T15:42:02+00:00"
        },
```
`

**Comment by dvdoug** [comment]
EPSG have released a new version of the dataset today which includes the correct boundary for Lambert-93 - I will try to get an updated PHPCoord release done this week that includes it.
